### ISSUE TITLE
Min / max indexes are not adjusted on prepend / append 

### ISSUE THREAD

**Issue by arudnev** [angular-ui/ui-scroll#99]
When prepend / append operations are called on the adapter it does not result into changes of buffer.minIndex / buffer.maxIndex (setLower / setUpper are not called), which leads to incorrect calculation of top and bottom paddings in adjustPadding operation when datasource.minIndex / datasource.maxIndex are provided / maintained.

The following change seems to be fixing the issue:

```javascript
    this.append = function (newItems) {
      buffer.append(newItems);
      buffer.setUpper();
      adjustBuffer();
    };

    this.prepend = function (newItems) {
      buffer.prepend(newItems);
      buffer.setLower();
      adjustBuffer();
    };
```

**Comment by mfeingold** [comment]
makes sense. Can you submit a PR? And if it not too much to ask - can you extend the test coverage to cover your use case(s)?

**Comment by dhilt** [comment]
This seems reasonable to adjust buffer min/max indexes each time the buffuer append/prepend is called, so I pushed some changes to 'master'. Btw the repro of initial issue would be very desirable for I couldn't catch the problem easily...

**Comment by arudnev** [comment]
Sorry for not getting back sooner.

I've added update of datasource minIndex / maxIndex to append / prepend sync demo, you can see the result here - http://rawgit.com/armortext/ui-scroll/master/demo/append/append.html

The changed code is here:

```javascript
		$scope.prepend = function () {
			var newItem = Server.prependItem(' (new)*');
			$scope.datasource.minIndex = Server.first - Server.prependedData.length;
			if (bof) {
				$scope.adapter.prepend([newItem]);
			}
		};

		$scope.append = function () {
			var newItem = Server.appendItem(' (new)*');
			$scope.datasource.maxIndex = Server.max + Server.appendedData.length;
			if (eof) {
				$scope.adapter.append([newItem]);
			}
		};
```

I most probably will not be able to add to test cases in the nearest days, but this should be enough for one time manual test.

**Comment by dhilt** [comment]
Since we have released v1.5.0 this issue becames not so actual. Also you may try the latest code with the fix you suggested before. Is it ok in your opinion? The end behaviour is not ideal I guess but much better than in your demo on v1.4.1...

**Comment by arudnev** [comment]
I've just merged the latest code from master and the issue is gone. But I believe I was using post-1.5.0 sources / dist when I ran into it. When I install 1.5.0 from via bower I get the following in the dist/ui-scroll.js headers - Version: 1.4.1 -- 2016-06-02T19:39:46.238Z and I think that's what I was using in the sample to reproduce before I merged.
Unless you meant that you just released 1.5.1, let me check...
